doc/api/url.md

```diff
@@ -250,7 +250,464 @@ properties of URL objects:
 For example, the ASCII space character (`' '`) is encoded as `%20`. The ASCII
 forward slash (`/`) character is encoded as `%3C`.
 
+## The WHATWG URL API
+
+> Stability: 1 - Experimental
+
+The `url` module provides an *experimental* implementation of the
+[WHATWG URL Standard][] as an alternative to the existing `url.parse()` API.
+
+```js
+const URL = require('url').URL;
+const myURL = new URL('https://example.org/foo');
+
+console.log(myURL.href);     // https://example.org/foo
+console.log(myURL.protocol); // https:
+console.log(myURL.hostname); // example.org
+console.log(myURL.pathname); // /foo
+```
+
+### Class: URL
+#### Constructor: new URL(input[, base])
+
+* `input` {String} The input URL to parse
+* `base` {String | URL} The base URL to resolve against if the `input` is not
```

Comment:
Ok, just verified, the code is doing the right thing here for anything that can reasonably be coerced into a string via toString. Passing `Url` instances as produced by `url.parse()` does not work because those do not implement a `toString()`.